inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImpl.java

```diff
@@ -285,7 +296,17 @@ private void clearState(String aUsername)
             states.remove(aUsername);
         }
     }
-        
+
+    private void clearPredictionsForProject(String aUsername, Project project)
+    {
+        synchronized (states) {
+            RecommendationUserState state = getState(aUsername);
+            state.clearActivePredictions(project);
+            state.clearIncomingPredictions(project);
+            state.clearTrainedModels();
+            state.clearActiveRecommenders();
```

Comment:
Do we need such fine-grained control over the clearing? Would a `state.clear(project)` not suffice?

Reply:
The RecommendationUserState expects that the user is only actively working on one project at a time. IMHO it would be sufficient to clear/rebuild the entire recommender state when a document is reset. But I thing having one method call instead of four would be sufficient - or do we need fine-grained control over which parts of the state are cleared? I included the "project" parameter because you had it in `state.clearActivePredictions(project);`.

Reply:
I would like to avoid keeping recommender states for multiple projects in memory because they would easily fill up memory.

Reply:
Ok, addressed in latest commit.